cmd/kube-scheduler/app/options/configfile.go

```diff
@@ -36,10 +39,39 @@ func loadConfigFromFile(file string) (*kubeschedulerconfig.KubeSchedulerConfigur
 	return loadConfig(data)
 }
 
+// newLenientSchemeAndCodecs returns a scheme that has only v1beta1 registered into
+// it and a CodecFactory with strict decoding disabled.
+func newLenientSchemeAndCodecs() (*runtime.Scheme, *serializer.CodecFactory, error) {
+	lenientScheme := runtime.NewScheme()
+	if err := kubeschedulerconfig.AddToScheme(lenientScheme); err != nil {
+		return nil, nil, fmt.Errorf("failed to add kubelet config API to lenient scheme: %v", err)
+	}
+	if err := kubeschedulerconfigv1alpha1.AddToScheme(lenientScheme); err != nil {
+		return nil, nil, fmt.Errorf("failed to add kubelet config v1alpha1 API to lenient scheme: %v", err)
+	}
+	lenientCodecs := serializer.NewCodecFactory(lenientScheme, serializer.DisableStrict)
+	return lenientScheme, &lenientCodecs, nil
+}
+
 func loadConfig(data []byte) (*kubeschedulerconfig.KubeSchedulerConfiguration, error) {
 	configObj := &kubeschedulerconfig.KubeSchedulerConfiguration{}
-	if err := runtime.DecodeInto(kubeschedulerscheme.Codecs.UniversalDecoder(), data, configObj); err != nil {
-		return nil, err
+	// The UniversalDecoder runs defaulting and returns the internal type by default.
+	err := runtime.DecodeInto(kubeschedulerscheme.Codecs.UniversalDecoder(), data, configObj)
+	if err != nil {
+		// Try strict decoding first. If that fails decode with a lenient
+		// decoder, which has only v1beta1 registered, and log a warning.
+		// The lenient path is to be dropped when support for v1alpha1 is dropped.
+		if !runtime.IsStrictDecodingError(err) {
+			return nil, err
+		}
+		klog.Warningf("using lenient decoding as strict decoding failed: %v", err)
+		_, lenientCodecs, err := newLenientSchemeAndCodecs()
+		if err != nil {
+			return nil, err
+		}
+		if err := runtime.DecodeInto(lenientCodecs.UniversalDecoder(), data, configObj); err != nil {
+			return nil, err
+		}
```

Comment:
This is turning into a pattern of some sorts. It might be good to extract it as an utility func in  some common place.

Reply:
I put it into `component-base` for a start: https://github.com/kubernetes/kubernetes/blob/1ed5ea3f2010916a62188d5d1948255a15a9b613/staging/src/k8s.io/component-base/codec/codec.go#L32

happy to discuss if / where it makes sense.